lib/Workflow/Condition.pm

```diff
@@ -25,6 +29,140 @@ sub evaluate {
     croak "Class ", ref($self), " must implement 'evaluate()'!\n";
 }
 
+
+sub evaluate_condition {
+    my ( $class, $wf, $condition_name) = @_;
+    $log ||= get_logger();
+    $wf->type;
+
+    my $factory = $wf->_factory();
+    my $orig_condition = $condition_name;
+    my $opposite       = 0;
+    my $condition;
+
+    $log->is_debug
+        && $log->debug("Checking condition $condition_name");
+
+    if ( $condition_name =~ m{ \A ! }xms ) {
+
+        # this condition starts with a '!' and is thus supposed
+        # to return the opposite of an original condition, whose
+        # name is the same except for the '!'
+        $orig_condition =~ s{ \A ! }{}xms;
+        $opposite = 1;
```

Comment:
What about renaming this to negation

```suggestion
        $negation = 1;
```

Reply:
Yup. Good idea. I found the wording confusing too, but since the code has been moved between files, I stuck to it for ease of comparison. It's much better this way.